backends/bmv2/common/parser.cpp

```diff
@@ -73,53 +73,83 @@ Util::IJson* ParserConverter::convertParserStatement(const IR::StatOrDecl* stat)
+                            if (parentType->is<IR::Type_Stack>()) {
+                                // stack.next.unionfield
+                                if (parent->member == IR::Type_Stack::next) {
+                                    type = "union_stack";
+                                    j = ctxt->conv->convert(parent->expr);
+                                    Util::JsonArray *a;
+                                    if (j->is<Util::JsonArray>()) {
+                                        a = j->to<Util::JsonArray>()->clone();
+                                    } else if (j->is<Util::JsonObject>()) {
+                                        a = new Util::JsonArray();
+                                        a->push_back(j->to<Util::JsonObject>()->get("value"));
+                                    } else {
+                                        BUG("unexpected");
+                                    }
+                                    a->append(mem->member.name);
+                                    auto j0 = new Util::JsonObject();
+                                    j = j0->emplace("value", a);
+                                } else {
+                                    BUG("%1%: unsupported", mem);
+                                }
                             }
                         }
                     }
-                    if (j == nullptr) {
-                        type = "regular";
-                        j = ctxt->conv->convert(arg->expression);
-                    }
-                    auto value = j->to<Util::JsonObject>()->get("value");
-                    param->emplace("type", type);
-                    param->emplace("value", value);
-
-                    if (argCount == 2) {
-                        auto arg2 = mce->arguments->at(1);
-                        auto jexpr = ctxt->conv->convert(arg2->expression, true, false);
-                        auto rwrap = new Util::JsonObject();
-                        // The spec says that this must always be wrapped in an expression
-                        rwrap->emplace("type", "expression");
-                        rwrap->emplace("value", jexpr);
-                        params->append(rwrap);
-                    }
-                    return result;
                 }
+                if (j == nullptr) {
+                    type = "regular";
+                    j = ctxt->conv->convert(arg->expression);
+                }
+                auto value = j->to<Util::JsonObject>()->get("value");
+                param->emplace("type", type);
+                param->emplace("value", value);
+
+                if (argCount == 2) {
+                    auto arg2 = mce->arguments->at(1);
+                    auto jexpr = ctxt->conv->convert(arg2->expression, true, false);
+                    auto rwrap = new Util::JsonObject();
+                    // The spec says that this must always be wrapped in an expression
```

Comment:
I thought `convert` could do that for you already (`wrap` parameter)?
So maybe if you call `auto jexpr = ctxt->conv->convert(arg2->expression, true, true);` you won't need to wrap yourself?